Objects/listobject.c

```diff
@@ -495,10 +506,9 @@ list_slice(PyListObject *a, Py_ssize_t ilow, Py_ssize_t ihigh)
         return NULL;
 
     src = a->ob_item + ilow;
-    dest = np->ob_item;
     for (i = 0; i < len; i++) {
         PyObject *v = src[i];
-        dest[i] = Py_NewRef(v);
+        _Py_SET_ITEMREF(np, i, v);
```

Comment:
The original code was okay -- we don't need an atomic store here because `dest[i]` is in a newly allocated list.